packages/gatsby/src/utils/page-data.js

```diff
@@ -1,26 +1,29 @@
 const fs = require(`fs-extra`)
 const path = require(`path`)
 const { store } = require(`../redux`)
+const { getPageHtmlFilePath } = require(`../utils/page-html`)
+
+const fixedPagePath = pagePath => (pagePath === `/` ? `index` : pagePath)
+
+const getPageDataFilePath = ({ publicDir }, pagePath) =>
+  path.join(publicDir, `page-data`, fixedPagePath(pagePath), `page-data.json`)
 
-const getFilePath = ({ publicDir }, pagePath) => {
-  const fixedPagePath = pagePath === `/` ? `index` : pagePath
-  return path.join(publicDir, `page-data`, fixedPagePath, `page-data.json`)
-}
 const read = async ({ publicDir }, pagePath) => {
-  const filePath = getFilePath({ publicDir }, pagePath)
+  const filePath = getPageDataFilePath({ publicDir }, pagePath)
   const rawPageData = await fs.readFile(filePath, `utf-8`)
   return JSON.parse(rawPageData)
 }
 
 const write = async ({ publicDir }, page, result) => {
-  const filePath = getFilePath({ publicDir }, page.path)
+  const filePath = getPageDataFilePath({ publicDir }, page.path)
```

Comment:
Let's avoid unnecessary code changes - this doesn't change anything and just make git history harder to follow.

I get that `getFilePath` vs `getPageDataFilePath` make things easier to understand - but this is only because we have lot of utilities not strictly related to `page-data` now. I think there is some code organization needed because we are leaking concerns.

Will comment on what exactly I think would be good places for things once I figure it out ;)

Reply:
@pieh

I've reverted `getPageDataFilePath ` back to using `getFilePath`. I've also created a `remove` function in both `utils/page-data.js` and `utils/page-html.js` and reused them.
 
Just waiting on where we should move these functions 🙇 

- `getChangedPageDataKeys`
 - `collectRemovedPageData`
 - ` removePageFiles`